> Phân tích sơ lược chỉ số giá tiêu dùng (CPI) Việt Nam

#### Coding setup....

## 1. Dữ liệu và tham khảo

* Wiki: [Chỉ số giá tiêu dùng (CPI) là gì?](https://vi.wikipedia.org/wiki/Ch%E1%BB%89_s%E1%BB%91_gi%C3%A1_ti%C3%AAu_d%C3%B9ng)
* World Bank: [data.worldbank.org VN 1995-2020](https://data.worldbank.org/indicator/FP.CPI.TOTL?end=2020&locations=VN&start=1995&view=chart)

### Sơ lược dữ liệu CPI từ World Bank

In [3]:
df_wb_cpi = pd.read_excel('https://api.worldbank.org/v2/en/indicator/FP.CPI.TOTL?downloadformat=excel', header = 3)
df_wb_cpi.columns

Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004',
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014', '2015', '2016', '2017', '2018', '2019', '2020'],
      dtype='object')

In [4]:
df_data = df_wb_cpi[["Country Name", "Country Code", "Indicator Name", "1960", "1990", "2000", "2010", "2020"]].copy()
df_data.tail()

Country Name Country Code                     Indicator Name  1960  1990  \
261        Kosovo          XKX  Consumer price index (2010 = 100)   NaN   NaN   
262   Yemen, Rep.          YEM  Consumer price index (2010 = 100)   NaN  3.84   
263  South Africa          ZAF  Consumer price index (2010 = 100)  1.77 25.43   
264        Zambia          ZMB  Consumer price index (2010 = 100)   NaN  0.22   
265      Zimbabwe          ZWE  Consumer price index (2010 = 100)   NaN   NaN   

     2000   2010   2020  
261   NaN 100.00 118.30  
262 34.24 100.00    NaN  
263 59.90 100.00 164.05  
264 23.83 100.00 245.71  
265   NaN 100.00    NaN

* Consumer price index (**2010 = 100**)
  * Nghĩa là giá tiêu dùng năm cơ sở được lấy là năm 2010, tỷ lệ = 100%
* Ví dụ **South Africa** năm 2020 CPI = 164
  * Nghĩa là giá tiêu dùng năm 2020 tăng 164% ~ 1.64 so với năm 2010
* Thông tin gồm: Country Name (quốc gia) và số liệu từ năm 1960 tới năm 2020
  * Tổng cộng có dữ liệu khoảng 265 quốc gia
* NaN: là những năm không có dữ liệu

## 2. Chỉ số giá tiêu dùng (CPI) Việt Nam

In [5]:
df_vnm = df_wb_cpi[df_wb_cpi["Country Code"] == "VNM"].copy()
df_vnm = df_vnm.dropna(axis = 1)
df_vnm

Country Name Country Code                     Indicator Name  \
257      Vietnam          VNM  Consumer price index (2010 = 100)   

    Indicator Code  1995  1996  1997  1998  1999  2000  ...   2011   2012  \
257    FP.CPI.TOTL 40.17 42.45 43.81 46.99 48.93 48.09  ... 118.68 129.47   

      2013   2014   2015   2016   2017   2018   2019   2020  
257 138.01 143.64 144.55 148.41 153.63 159.07 163.52 168.78  

[1 rows x 30 columns]

* Dữ liệu chỉ số giá tiêu dùng VietNam chỉ có từ năm **1995** tới năm **2020**

In [6]:
col_remap = dict(zip(df_wb_cpi.T.columns, df_wb_cpi.T.loc["Country Code"]))

dft_vnm = df_vnm.T[4:]
dft_vnm.rename(columns = col_remap, inplace = True)
dft_vnm.T

1995  1996  1997  1998  1999  2000  2001  2002  2003  2004  ...   2011  \
VNM 40.17 42.45 43.81 46.99 48.93 48.09 47.88 49.72 51.32 55.30  ... 118.68   

      2012   2013   2014   2015   2016   2017   2018   2019   2020  
VNM 129.47 138.01 143.64 144.55 148.41 153.63 159.07 163.52 168.78  

[1 rows x 26 columns]

In [7]:
dft_vnm.iplot(kind="line", title = "VietNam CPI - 2020, World Bank"
              , yTitle = "X% vs 2010", xTitle = "Years")

### Sơ lược:
* Từ năm 2010 tới năm 2020 chỉ số giá tiêu dùng tăng 1.68 (168%).
* Năm 2000 cpi = 48% so với năm 2010, nghĩa là giá tiêu dùng cơ bản tăng gấp đôi qua 10 năm
* Năm 1995 cpi = 40% so với năm 2020 cpi = 160%, giá tiêu dùng tăng 4 lần kể từ năm 1995.

### Câu hỏi nghiên cứu
* ~~2.1. CPI Việt Nam so với~~ Mỹ, Trung Quốc, Pháp, Đan Mạch, và hàng xóm Singapore, Thái Lan có khác nhau?
* 2.2. Yếu tố nào có biến động tương đương, bằng hoặc cao hơn CPI từ năm 1995 tới năm 2020?
  * Vàng, bất động sản?
  * Chứng khoán, cố phiếu?
  * Bảo hiểm? Bảo hiểm xã hội?
  * Lãi suất ngân hàng (gửi tiết kiệm)?
  * GDP, PPP
* 2.3. Một cái Tivi, Tủ lạnh, hoặc hàng tiêu dùng giá 1 triệu năm 1995 so với năm 2020 có gấp 4 lần như chỉ số CPI từ World Bank?
* 2.4. CPI Việt Nam theo cách nhóm hàng tiêu dùng có khác nhau nhiều từ 1995?

## 3. So sánh CPI Vietnam và các nước

> Trả lời câu hỏi 2.1

Để thấy được tương quan CPI giữa Việt Nam và các nước, chúng ta sẽ thử so sánh CPI VN với một vài nước đại diện sau đây:
* Bắc Mỹ: Mỹ
* Tây Âu: Pháp
* Bắc Âu: Đan Mạch
* Tây Á : UAE (Ả Rập)
* Châu Á: Trung Quốc, Hàn Quốc
* Hàng xóm: Thái Lan, Singapore, và Lào


In [8]:
country_codes = ['VNM', 'USA', 'FRA', 'DNK', 'ARE', 'CHN', 'KOR', 'THA', 'SGP', 'LAO']
col_remap = dict(zip(df_wb_cpi.T.columns, df_wb_cpi.T.loc["Country Code"]))

In [9]:
df_cpi = df_wb_cpi[df_wb_cpi["Country Code"].isin(country_codes)].copy()
# df_cpi = df_cpi.dropna(axis = 1)

dft_cpi = df_cpi.T[4:]
dft_cpi.rename(columns = col_remap, inplace = True)
dft_cpi.head()

ARE  CHN   DNK   FRA  KOR  LAO   SGP   THA   USA  VNM
1960  NaN  NaN  8.25 10.43 1.48  NaN 27.52 10.21 13.56  NaN
1961  NaN  NaN  8.53 10.68 1.60  NaN 27.63 10.97 13.71  NaN
1962  NaN  NaN  9.16 11.25 1.70  NaN 27.75 11.37 13.87  NaN
1963  NaN  NaN  9.72 11.82 2.06  NaN 28.36 11.37 14.04  NaN
1964  NaN  NaN 10.02 12.20 2.66  NaN 28.85 11.28 14.22  NaN

In [10]:
dft_cpi.iplot(kind="line", title = "CPI - 2020, World Bank"
              , yTitle = "X% vs 2010", xTitle = "Years")

### Sơ lược
* Chỉ số CPI Việt Nam nằm **top bảng** 160% năm 2020, cao hơn Lào 142%, Trung Quốc 128%. Nhóm các quốc gia còn lại CPI chỉ giao động dưới 120% trong 10 năm qua từ năm 2010.
* Từ năm 1990 tới năm 2010, **Lào** là nước có bước tăng trưởng CPI mạnh mẽ nhất từ 1.67% lên 100% trong 20 năm. Ở góc độ tiêu cực thì có vẻ là lạm phát 50 lần, ở góc độ tích cực thì chất lượng cuộc sống (giỏ hàng) tăng 50 lần.

### Câu hỏi nghiên cứu:
* 3.1. Nguyên nhân nào khiến chỉ số CPI Việt Nam khá cao so với thế giới? 
  * Do mệnh giá VND thấp?
  * Do tăng trưởng GDP hàng năm cao? Việt Nam đang trên đà bứt ra khỏi nhóm nước đang phát triển?
  * Do giỏ hàng tiêu dùng của người Việt Nam dịch chuyển sang nhóm hàng có giá trị cao hơn?
* 3.2. Con số CPI tuyệt đối từ World Bank là bao nhiêu?
* 3.3. Sự gia tăng đáng kể CPI ở Lào và Việt Nam từ năm 1990 là do lạm phát hay do sự dịch chuyển trong cấu trúc giỏ hàng tiêu dùng theo số liệu từ World Bank?